src/utils/namespace-context-utils.ts

```diff
@@ -2,17 +2,24 @@ import * as React from 'react';
 import { k8sListResourceItems } from '@openshift/dynamic-plugin-sdk-utils';
 import { ProjectModel } from '../models';
 
-type NamespaceContextData = { namespace: string; namespaceLoaded: boolean };
+type NamespaceContextData = {
+  namespace: string;
+  namespaceLoaded: boolean;
+  setActiveNamespace: React.Dispatch<React.SetStateAction<string>>;
+};
 
 export const NamespaceContext = React.createContext<NamespaceContextData>({
   namespace: '',
   namespaceLoaded: false,
+  setActiveNamespace: () => {},
 });
 
 export const NamespaceProvider = NamespaceContext.Provider;
 
 export const useNamespace = () => React.useContext(NamespaceContext).namespace;
 
+export const useSetNamespace = () => React.useContext(NamespaceContext).setActiveNamespace;
+
 export const useActiveNamespace = (): NamespaceContextData => {
```

Comment:
We shouldn't be using this implementation to set the namespace anymore. We have much more reliable data coming from workspace lister API. I think we should just set the namespace context directly from there. No need to set it using this method. This can become error prone if there's a race condition between setting workspace context and fetching namespace from that workspace.